lib/conditional.module.ts

```diff
@@ -0,0 +1,24 @@
+import { ModuleMetadata } from '@nestjs/common';
+import { ConfigModule } from '.';
+
+export class ConditionalModule {
+  static async registerWhen(
+    module: Required<ModuleMetadata>['imports'][number],
+    condition: string | ((env: NodeJS.ProcessEnv) => boolean),
+  ) {
```

Comment:
Do we want to add in an optional third parameter with a `setTimeout` or similar that checks if we've resolved by the elapsed time or throws an error? Just so the application doesn't hang while waiting for the config variables to load (essentially if the `ConfigModule.forRoot()` is never called)

Reply:
Makes sense yeah. I think we can add a third `options` parameter with a `timeout` attribute so if we ever need to add more props then we can just add them to the opts object

Reply:
Done. Set to 5 seconds for the moment